### DISCUSSION TITLE
Github Actions failing everyday with this error, Kindly help

### DISCUSSION THREAD

**Discussion by rexdivakar** [lowlighter/metrics#402] [Q&A] [answered]
<html>
<body>
<div class="mt-1 mb-3 px-3" style="box-sizing: border-box; margin-top: 4px !important; margin-bottom: 16px !important; padding-left: 16px !important; padding-right: 16px !important; color: rgb(36, 41, 46); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="mx-0 mx-md-1" style="box-sizing: border-box; margin-left: 4px !important; margin-right: 4px !important;"><h4 class="text-bold" style="box-sizing: border-box; margin-bottom: 0px; margin-top: 0px; font-weight: 600 !important; font-size: 16px;">Annotations</h4><div class="color-text-secondary text-small" style="box-sizing: border-box; color: var(--color-text-secondary) !important; font-size: 12px !important;">1 warning</div></div></div>

github-metricsUnexpected input(s) 'plugin_habits_trim', valid inputs are ['base', 'repositories', 'repositories_forks', 'repositories_affiliations', 'repositories_skipped', 'commits_authoring', 'token', 'user', 'repo', 'committer_token', 'committer_branch', 'committer_message', 'committer_gist', 'filename', 'markdown', 'markdown_cache', 'output_action', 'optimize', 'setup_community_templates', 'template', 'query', 'extras_css', 'config_timezone', 'config_order', 'config_twemoji', 'config_gemoji', 'config_display', 'config_animations', 'config_padding', 'config_output', 'retries', 'retries_delay', 'plugins_errors_fatal', 'debug', 'verify', 'debug_flags', 'dryrun', 'experimental_features', 'use_mocked_data', 'use_prebuilt_image', 'plugin_isocalendar', 'plugin_isocalendar_duration', 'plugin_languages', 'plugin_languages_ignored', 'plugin_languages_skipped', 'plugin_languages_limit', 'plugin_languages_sections', 'plugin_languages_colors', 'plugin_languages_details', 'plugin_languages_threshold', 'plugin_languages_indepth', 'plugin_languages_recent_load', 'plugin_languages_recent_days', 'plugin_introduction', 'plugin_introduction_title', 'plugin_habits', 'plugin_habits_from', 'plugin_habits_days', 'plugin_habits_facts', 'plugin_habits_charts', 'plugin_licenses', 'plugin_licenses_setup', 'plugin_licenses_ratio', 'plugin_licenses_legal', 'plugin_contributors', 'plugin_contributors_base', 'plugin_contributors_head', 'plugin_contributors_ignored', 'plugin_contributors_contributions', 'plugin_stars', 'plugin_stars_limit', 'plugin_topics', 'plugin_topics_mode', 'plugin_topics_sort', 'plugin_topics_limit', 'plugin_reactions', 'plugin_reactions_limit', 'plugin_reactions_days', 'plugin_reactions_display', 'plugin_reactions_details', 'plugin_reactions_ignored', 'plugin_people', 'plugin_people_limit', 'plugin_people_size', 'plugin_people_types', 'plugin_people_thanks', 'plugin_people_sponsors_custom', 'plugin_people_identicons', 'plugin_people_shuffle', 'plugin_gists', 'plugin_followup', 'plugin_followup_sections', 'plugin_lines', 'plugin_lines_skipped', 'plugin_traffic', 'plugin_stargazers', 'plugin_projects', 'plugin_projects_limit', 'plugin_projects_repositories', 'plugin_projects_descriptions', 'plugin_activity', 'plugin_activity_limit', 'plugin_activity_load', 'plugin_activity_days', 'plugin_activity_filter', 'plugin_activity_visibility', 'plugin_activity_timestamps', 'plugin_activity_skipped', 'plugin_achievements', 'plugin_achievements_threshold', 'plugin_achievements_secrets', 'plugin_achievements_limit', 'plugin_achievements_ignored', 'plugin_achievements_only', 'plugin_notable', 'plugin_notable_filter', 'plugin_notable_repositories', 'plugin_support', 'plugin_skyline', 'plugin_skyline_year', 'plugin_skyline_frames', 'plugin_skyline_quality', 'plugin_skyline_compatibility', 'plugin_pagespeed', 'plugin_pagespeed_url', 'plugin_pagespeed_detailed', 'plugin_pagespeed_screenshot', 'plugin_pagespeed_token', 'plugin_music', 'plugin_music_provider', 'plugin_music_token', 'plugin_music_mode', 'plugin_music_playlist', 'plugin_music_limit', 'plugin_music_played_at', 'plugin_music_user', 'plugin_tweets', 'plugin_tweets_token', 'plugin_tweets_attachments', 'plugin_tweets_limit', 'plugin_tweets_user', 'plugin_anilist', 'plugin_anilist_medias', 'plugin_anilist_sections', 'plugin_anilist_limit', 'plugin_anilist_limit_characters', 'plugin_anilist_shuffle', 'plugin_anilist_user', 'plugin_stackoverflow', 'plugin_stackoverflow_user', 'plugin_stackoverflow_sections', 'plugin_stackoverflow_limit', 'plugin_stackoverflow_lines', 'plugin_stackoverflow_lines_snippet', 'plugin_posts', 'plugin_posts_source', 'plugin_posts_descriptions', 'plugin_posts_covers', 'plugin_posts_limit', 'plugin_posts_user', 'plugin_rss', 'plugin_rss_source', 'plugin_rss_limit', 'plugin_wakatime', 'plugin_wakatime_token', 'plugin_wakatime_days', 'plugin_wakatime_sections', 'plugin_wakatime_limit', 'plugin_wakatime_url', 'plugin_wakatime_user', 'plugin_stock', 'plugin_stock_token', 'plugin_stock_symbol', 'plugin_stock_duration', 'plugin_stock_interval', 'plugin_screenshot', 'plugin_screenshot_title', 'plugin_screenshot_url', 'plugin_screenshot_selector', 'plugin_screenshot_background', 'plugin_nightscout', 'plugin_nightscout_url', 'plugin_nightscout_datapoints', 'plugin_nightscout_lowalert', 'plugin_nightscout_highalert', 'plugin_nightscout_urgentlowalert', 'plugin_nightscout_urgenthighalert']
--

</body>
</html>

**Comment by lowlighter** [comment, marked as answer]
Hello!

This is just a warning because you used an unsupported option `plugin_habits_trim` 🙂 

The actual cause from https://github.com/rexdivakar/rexdivakar/runs/2920518636?check_suite_focus=true#step:2:320 :
```
GitHub token                                             │ (provided)
...
RequestError [HttpError]: Unauthorized
────────────────────────────────────────────────────────────────────────────────────────
undefined
An error occured, logging debug message :
    at /metrics/node_modules/@octokit/request/dist-node/index.js:51:13
    at processTicksAndRejections (node:internal/process/task_queues:94:5)
    at async file:///metrics/source/app/action/index.mjs:131:25 {
  status: 401,
```

It usually means that your token is not valid, maybe try to regenerate your token and set it again in your repositories secret 👍

**Reply by Zenoo** [reply]
As a side note, `plugin_habits_trim` is an actual option on master. But it hasn't been released yet.

**Reply by rexdivakar** [reply]
Got it, now its fixed with new token update Thank you
